### ISSUE TITLE
DS4 Latency over Wire

### ISSUE THREAD

**Issue by Sxregii** [Ryochan7/DS4Windows#1257]
Hi, i've recently changed my polling rate to 1000mhz to get a constant low input delay but when i launch ds4 windows, the input delay spikes up to over 10ms which feezes my controller for a second. This happens every 6/5 seconds and it really frustrating.

**Comment by mika-n** [comment]
Latency over wire? Or do you mean wireless (bluetooth)?

USB connection should have a steady stream of bits no matter what the polling rate value in DS4Windows settings is (ie. the polling rate config option is used only with BT connections).

**Comment by Sxregii** [comment]
Nope, i mean by latency over USB connection. If you want you can add me on discord or something and i could share my screen and show you how the issue works. Sxregii#9315 on Discord.

**Comment by mika-n** [comment]
I have heard only one or two cases where USB connection have had latency. The throughput of USB channel should be moooore than enough for DS4 gamepads. At least in one case the issue turn out to be faulty usb cable/port, so try different cable and USB port. And remove all other usb devices to see if it makes a difference.

**Comment by Sxregii** [comment]
I've checked my ports and used 3 USB cables. Still the same results. Sorry for late reply.
________________________________

**Comment by Rogus** [comment]
> I have heard only one or two cases where USB connection have had latency. The throughput of USB channel should be moooore than enough for DS4 gamepads. At least in one case the issue turn out to be faulty usb cable/port, so try different cable and USB port. And remove all other usb devices to see if it makes a difference.

So he has to try without the other USB ports like keyboard and mouse if he uses of course. I would love to try reproduce the issue. But I have only a mouse with a USB connection and a keyboard USB.

**Comment by Sxregii** [comment]
yep just double checked if that would work but it didnt.

**Comment by mika-n** [comment]
If cable and usb port issue has been ruled out then I'm running out of ideas. Maybe if you have an access to another gamepad or PC/laptop then try those vice-versa to test if the issue is gone when the gamepad or PC is "changed".

**Comment by kurisu-gh** [comment]
Did you happen to use this tool to increase the poll rate? https://github.com/LordOfMice/hidusbf

What happens if you revert back to normal?

**Comment by Sxregii** [comment]
Yep thats the tool

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by kurisu-gh** [comment]
I tried the same tool several months back and I think I experienced the same problems with DS4windows. I stopped using hidusbf to get rid of the problem, and I wasn't a fan of having to check if the modifications remained after every system update.

Maybe there are some compatibility issues between the overclocked polling rate over USB and DS4windows but I'm not sure. Perhaps @Ryochan7 or @mika-n might know?

**Comment by silverxd** [comment]
I have the exact same issue. Some time ago I opened an issue about a very similar problem, only for me the latency was appearing over Bluetooth only. Well not anymore. Whenever I start up DS4Windows in any mode or config, my input delay instantly spikes from an unnoticeable ~5ms to about 500-1000ms. This now happens in both wired and wireless connections. Using the exact same gadgets to connect without DS4Windows- zero noticeable input lag. I tested it using GTA V, CS:GO (yes I play cs with a controller), Fortnite and multiple controller testing applications. All report the same issue. However the input delay seems to not be there in DS4 joystick preview screen. The delay seems to appear between DS4Windows and the game. "Hide controller" changed nothing. Latest windows, latest ds4windows. It is indeed very weird, I'm going to try a complete reinstall of DS4Windows later tomorrow. 
Best Regards 
Silver
